python/cudf/cudf/tests/test_datetime.py

```diff
@@ -81,6 +81,11 @@ def numerical_data():
     "hour",
     "minute",
     "second",
+    "microsecond",
+    # Pandas supports 'second', 'microsecond' & 'nanosecond'
+    # but weirdly left out 'millisecond', hence can't have 'millisecond'
+    # in this list.
+    "nanosecond",
```

Comment:
The discrepancy I noted above in the behavior of `microsecond` was not caught in tests because the only tests ([`test_dt_series`](https://github.com/rapidsai/cudf/blob/2c9de715a387692f319266f6c20ed8f9525a1fc5/python/cudf/cudf/tests/test_datetime.py#L182), [`test_dt_index`](https://github.com/rapidsai/cudf/blob/2c9de715a387692f319266f6c20ed8f9525a1fc5/python/cudf/cudf/tests/test_datetime.py#L192)) using these [parameterized fields](https://github.com/rapidsai/cudf/blob/2c9de715a387692f319266f6c20ed8f9525a1fc5/python/cudf/cudf/tests/test_datetime.py#L77-L93) appear to [have frequency 400hr](https://github.com/rapidsai/cudf/blob/2c9de715a387692f319266f6c20ed8f9525a1fc5/python/cudf/cudf/tests/test_datetime.py#L26-L27) and thus the tests for sub-hour frequencies are probably not meaningful! We should improve the granularity of our test inputs so that we can have meaningful results for non-zero minutes, seconds, and smaller.

Reply:
Broadened the `data2` fixture. I don't know why we have duplicate fixture until now both `data1` and `data2` returning identical data 😕